webapp/components/signup/signup_controller.jsx

```diff
@@ -0,0 +1,318 @@
+                            id='signup.email'
+                            defaultMessage='Email and Password'
+                        />
+                    </span>
+                </Link>
+            );
+        }
+
+        if (this.state.gitlabEnabled) {
+            signupControls.push(
+                <a
+                    className='btn btn-custom-login btn--full gitlab'
+                    key='gitlab'
+                    href={Client.getOAuthRoute() + '/gitlab/signup' + window.location.search}
+                >
+                    <span className='icon'/>
+                    <span>
+                        <FormattedMessage
+                            id='signup.gitlab'
+                            defaultMessage='GitLab Single-Sign-On'
+                        />
+                    </span>
+                </a>
+            );
+        }
+
+        if (this.state.googleEnabled) {
+            signupControls.push(
+                <a
+                    className='btn btn-custom-login btn--full google'
+                    key='google'
+                    href={Client.getOAuthRoute() + '/google/signup' + window.location.search + '&team=' + encodeURIComponent(this.state.teamName)}
+                >
+                    <span className='icon'/>
+                    <span>
+                        <FormattedMessage
+                            id='signup.google'
+                            defaultMessage='Google Account'
+                        />
+                    </span>
+                </a>
+            );
+        }
+
+        if (this.state.office365Enabled) {
+            signupControls.push(
+                <a
+                    className='btn btn-custom-login btn--full office365'
+                    key='office365'
+                    href={Client.getOAuthRoute() + '/office365/signup' + window.location.search + '&team=' + encodeURIComponent(this.state.teamName)}
+                >
+                    <span className='icon'/>
+                    <span>
+                        <FormattedMessage
+                            id='signup.office365'
+                            defaultMessage='Office 365'
+                        />
+                    </span>
+                </a>
+           );
+        }
+
+        if (this.state.ldapEnabled) {
+            signupControls.push(
+                <Link
+                    className='btn btn-custom-login btn--full ldap'
+                    key='ldap'
+                    to={'/signup_ldap'}
+                >
+                    <span className='icon fa fa-folder-open fa--margin-top'/>
+                    <span>
+                        <FormattedMessage
+                            id='signup.ldap'
+                            defaultMessage='LDAP Credentials'
+                        />
+                    </span>
+                </Link>
+            );
+        }
+
+        if (this.state.samlEnabled) {
+            signupControls.push(
```

Comment:
@DavidLu1997 at least for SAML I think this is wrong

first of all you need an anchor tag instead of the *Link*  tag and you also need the parameters in case of signup

check at the code you're removing

```
<a
      className='btn btn-custom-login saml'
      key='saml'
      href={`/login/sso/saml${window.location.search}${window.location.search ? '&' : '?'}action=signup`}
>
      <span>
            {global.window.mm_config.SamlLoginButtonText}
       </span>
</a>
```

also note that the text for the button in SAML in configured in the System Console.